core-bundle/src/Resources/contao/library/Contao/Environment.php

```diff
@@ -296,9 +296,9 @@ protected static function httpHost()
 		}
 		else
 		{
-			$host = $_SERVER['SERVER_NAME'];
+			$host = $_SERVER['SERVER_NAME'] ?? null;
 
-			if ($_SERVER['SERVER_PORT'] != 80)
+			if (($_SERVER['SERVER_PORT'] ?? null) != 80)
```

Comment:
These env variables should always exist. Why did you have to change this?

Reply:
Sorry, I wasn't sure if it does - and it was close to the affected code `$_SERVER['SERVER_NAME']`. Shall I remove the check?

Reply:
Since other methods already use `System::getContainer()->get('request_stack')->getCurrentRequest()` may be the remaining methods should do the same? Then this issue would not occur.

Reply:
so the solution would be this? 

```php
	protected static function httpHost()
	{
		$request = System::getContainer()->get('request_stack')->getCurrentRequest();

		if ($request === null)
		{
			return null;
		}

		return $request->getHost();
	}
```

Reply:
You might still need to append the port manually though. Though I don't quite understand the old code. It appends the port if `HTTP_HOST` is empty and the `SERVER_PORT` is not `80`? That means it could return the port with an SSL connection?

Reply:
You're absolutely right I used the wrong method - it has to be `getHttpHost`, which I commited!

Reply:
I reverted the changes with Symfony request!
It is really also neccessary to check `SERVER_PORT` - because on CLI while cache clear it is not defined in my docker environment!